drivers/dma/dma_mcux_edma.c

```diff
@@ -634,10 +635,14 @@ static int dma_mcux_edma_reload(const struct device *dev, uint32_t channel,
 		/* Convert size into major loop count */
 		size = size / data->transfer_settings.dest_data_size;
 
+		/* Previous TCD index in circular list */
+		pre_idx = data->transfer_settings.write_idx - 1;
+		if (pre_idx >= CONFIG_DMA_TCD_QUEUE_SIZE)
+			pre_idx = CONFIG_DMA_TCD_QUEUE_SIZE - 1;
+
 		/* Configure a TCD for the transfer */
 		tcd = &(DEV_CFG(dev)->tcdpool[channel][data->transfer_settings.write_idx]);
-		pre_tcd = &(DEV_CFG(dev)->tcdpool[channel][(data->transfer_settings.write_idx - 1) %
```

Comment:
only modify this line is ok.
(data->transfer_settings.write_idx - 1 + CONFIG_DMA_TCD_QUEUE_SIZE) %
                                                           CONFIG_DMA_TCD_QUEUE_SIZE